Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2887,6 +2887,7 @@ protected void onPostExecute(ServerResponse serverResponse) {
                                 Log.i("BranchSDK", "Branch API Error: Conflicting resource error code from API");
                                 handleFailure(0, status);
                             }
+                            requestQueue_.remove(thisReq_);
```

Comment:
Would it be better to remove this before notifying the callback? The callback may want to queue a new request and if `thisReq_` is still at the head of the queue it may get repeated.